cores/esp8266/Schedule.cpp

```diff
@@ -132,37 +138,20 @@ bool schedule_recurrent_function_us(const std::function<bool(void)>& fn,
     }
     rLast = item;
 
-    // grain needs to be recomputed
-    recurrent_max_grain_mS = 0;
-
     return true;
 }
 
-uint32_t compute_scheduled_recurrent_grain ()
+uint32_t get_scheduled_recurrent_delay_us()
 {
-    if (recurrent_max_grain_mS == 0)
-    {
-        if (rFirst)
-        {
-            uint32_t recurrent_max_grain_uS = rFirst->callNow.getTimeout();
-            for (auto it = rFirst->mNext; it; it = it->mNext)
-                recurrent_max_grain_uS = std::gcd(recurrent_max_grain_uS, it->callNow.getTimeout());
-            if (recurrent_max_grain_uS)
-                // round to the upper millis
-                recurrent_max_grain_mS = recurrent_max_grain_uS <= 1000? 1: (recurrent_max_grain_uS + 999) / 1000;
-        }
-
-#ifdef DEBUG_ESP_CORE
-        static uint32_t last_grain = 0;
-        if (recurrent_max_grain_mS != last_grain)
-        {
-            ::printf(":rsf %u->%u\n", last_grain, recurrent_max_grain_mS);
-            last_grain = recurrent_max_grain_mS;
-        }
-#endif
-    }
+    if (!rFirst) return ~static_cast<decltype(micros())>(0) >> 1;
```

Comment:
`~static_cast<decltype(micros())>(0) >> 1` should be usefully replaced by `std::numeric_limits<decltype(micros())>::max()`

Why not using `using micros_t = decltype(micros())`
and `using micros_signed_t = std::make_signed_t<micros_t>` ?

Reply:
> `~static_cast<decltype(micros())>(0) >> 1` should be usefully replaced by `std::numeric_limits<decltype(micros())>::max()`

The results are dramatically different, unless also `>> 1`. I feel that the purpose becomes less "obvious" when working with "max", which had given thought and discarded due to that reason. The wrap-over logic is a very technical detail.

> Why not using `using micros_t = decltype(micros())` and `using micros_signed_t = std::make_signed_t<micros_t>` ?

I am really not a fan of aliasing. If your change is a prerequiste, re-request, please.

Reply:
> The wrap-over logic is a very technical detail.

Really?

The fact is that this code does not compile without warnings in 64 bit host mode.

How would your logic be different without `>>1` ?
Or from `rTarget = micros() + (~static_cast<decltype(micros())>(0) >> 1)` to `rTarget = micros() - 1`

I tried this and it seems to works with a trivial example.
Demixing `decltype(micros())` and `uint32` appears to be necessary.
```diff

--- a/cores/esp8266/Schedule.cpp
+++ b/cores/esp8266/Schedule.cpp
@@ -146,7 +146,7 @@ bool schedule_recurrent_function_us(const std::function<bool(void)>& fn,
 
 uint32_t get_scheduled_recurrent_delay_us()
 {
-    if (!rFirst) return ~static_cast<decltype(micros())>(0) >> 1;
+    if (!rFirst) return std::numeric_limits<uint32_t>::max();
     // handle already expired rTarget.
     const int32_t remaining = rTarget - micros();
     return (remaining > 0) ? static_cast<uint32_t>(remaining) : 0;
@@ -154,7 +154,7 @@ uint32_t get_scheduled_recurrent_delay_us()
 
 uint32_t get_scheduled_delay_us()
 {
-    return sFirst ? 0 : ~static_cast<decltype(micros())>(0) >> 1;
+    return sFirst ? 0 : std::numeric_limits<uint32_t>::max();
 }
 
 void run_scheduled_functions()
@@ -223,7 +223,7 @@ void run_scheduled_recurrent_functions()
 
     // prevent scheduling of new functions during this run
     stop = rLast;
-    rTarget = micros() + (~static_cast<decltype(micros())>(0) >> 1);
+    rTarget = micros() - 1;
```

Reply:
I would like to think that the new comment at HALF_MAX_MICROS explains everything. If it doesn't, let's improve the wording together.